src/utils/store.js

```diff
@@ -0,0 +1,193 @@
+        }
+    });
+};
+
+/**
+ * Will retrieve all data from the collection
+ * @returns {Promise}
+ */
+Store.prototype.findAll = function() {
+    return new Promise((resolve, reject) => {
+        try {
+            resolve(this.todos);
+        }
+        catch (err) {
+            reject(err);
+        }
+    });
+};
+
+/**
+ * Will save the given data to the DB. If no item exists it will create a new
+ * item, otherwise it'll simply update an existing item's properties
+ *
+ * @param {object} updateData The data to save back into the DB
+ * @param {number} id An optional param to enter an ID of an item to update
+ * @returns {Promise}
+ */
+Store.prototype.save = function(updateData, id) {
+    return new Promise((resolve, reject) => {
+        try {
+            let todos = this.todos;
+            // If an ID was actually given, find the item and update each property
+            if (id) {
+                for (let i = 0; i < todos.length; i++) {
+                    if (todos[i].id === id) {
+                        for (let key in updateData) {
+                            todos[i][key] = updateData[key];
+                        }
+                        break;
+                    }
+                }
+            } else {
+                // Generate a new id and add to store
+                updateData.id = new Date().getTime();
+                todos.push(updateData);
+            }
+            this.todos = todos;
+            resolve(updateData);
+        }
+        catch (err) {
+            reject(err);
+        }
+    });
+};
+
+/**
+ * Will remove an item from the Store based on its ID
+ *
+ * @param {number} id The ID of the item you want to remove
+ * @returns {Promise}
+ */
+Store.prototype.remove = function(id) {
+    return new Promise((resolve, reject) => {
+        try {
+            let todos = this.todos;
+            for (let i = 0; i < todos.length; i++) {
+                if (todos[i].id == id) {
+                    todos.splice(i, 1);
+                    break;
+                }
+            }
+            this.todos = todos;
+            resolve(todos);
+        } catch (err) {
+            reject(err);
+        }
+    });
+};
+
+/**
+ * Will clear all completed todos
+ *
+ * @returns {Promise}
+ */
+Store.prototype.ClearCompleted = function() {
+    return new Promise((resolve, reject) => {
+        try {
+            const todos = todos.filter(todo => {
+                if (todo.completed) {
+                    return false;
+                }
+                return true;
+            });
+            this.todos = todos;
+            resolve(todos);
+        } catch (err) {
+            reject(err);
+        }
+    });
+};
+
+/**
+ * Will toggle all todos
+ *
+ * @returns {Promise}
+ */
+Store.prototype.ToggleAll = function() {
+    return new Promise((resolve, reject) => {
+        try {
+            let todos = this.todos;
+            for (let i = 0; i < todos.length; i++) {
```

Comment:
using of `for` loops is old imperative approach, better to use declarative style.  
https://codeburst.io/declarative-vs-imperative-programming-a8a7c93d9ad2

here we can use `todos.forEach`